File: common/src/main/java/cn/rubintry/common/utils/http/BaseHttp.java
package cn.rubintry.common.utils.http;


import java.util.concurrent.TimeUnit;

/**
 * @author logcat
 */
public class BaseHttp {
    private String baseUrl;
    private Long connectTimeout;
    private Long readTimeout;
    private Long writeTimeout;

    public BaseHttp() {
    }

    public BaseHttp(String baseUrl) {
        this.baseUrl = baseUrl;
    }

    public String getBaseUrl() {
        return baseUrl;
    }

    public void setBaseUrl(String baseUrl) {
        this.baseUrl = baseUrl;
    }

    public Long getConnectTimeout() {
        return connectTimeout;
    }

    public void setConnectTimeout(Long connectTimeout) {
        setConnectTimeout(connectTimeout, TimeUnit.MILLISECONDS);
    }


    public void setConnectTimeout(Long connectTimeout, TimeUnit unit) {
        this.connectTimeout = getTimeOut(connectTimeout, unit);
    }

    public Long getReadTimeout() {
        return readTimeout;
    }

    public void setReadTimeout(Long readTimeout) {
        setReadTimeout(readTimeout, TimeUnit.MILLISECONDS);
    }

    public void setReadTimeout(Long readTimeout, TimeUnit unit) {
        this.readTimeout = getTimeOut(readTimeout, unit);
    }


    public Long getWriteTimeout() {
        return writeTimeout;
    }

    public void setWriteTimeout(Long writeTimeout) {
        setWriteTimeout(writeTimeout, TimeUnit.MILLISECONDS);
    }


    public void setWriteTimeout(Long writeTimeout, TimeUnit unit) {
        this.writeTimeout = getTimeOut(writeTimeout, unit);
    }

    private Long getTimeOut(Long timeout, TimeUnit unit) {
        long resultTime = 0;
        switch (unit) {
            case NANOSECONDS:
                resultTime = unit.convert(timeout, TimeUnit.NANOSECONDS);
                break;
            case MICROSECONDS:
                resultTime = unit.convert(timeout, TimeUnit.MICROSECONDS);
                break;
            case MILLISECONDS:
                resultTime = unit.convert(timeout, TimeUnit.MILLISECONDS);
                break;
            case SECONDS:
                resultTime = unit.convert(timeout, TimeUnit.SECONDS);
                break;
            case MINUTES:
                resultTime = unit.convert(timeout, TimeUnit.MINUTES);
                break;
            case HOURS:
                resultTime = unit.convert(timeout, TimeUnit.HOURS);
                break;
            case DAYS:
                resultTime = unit.convert(timeout, TimeUnit.DAYS);
                break;
            default:
                break;
        }
        return resultTime;
    }
}

File: common/src/main/java/cn/rubintry/common/utils/http/BaseHttp.kt
package cn.rubintry.common.utils.http

import java.util.concurrent.TimeUnit

/**
 * @author logcat
 */
class BaseHttp {
    var baseUrl: String? = null
    private var connectTimeout: Long? = null
    private var readTimeout: Long? = null
    private var writeTimeout: Long? = null

    constructor() {}
    constructor(baseUrl: String?) {
        this.baseUrl = baseUrl
    }

    fun getConnectTimeout(): Long? {
        return connectTimeout
    }

    fun setConnectTimeout(connectTimeout: Long?) {
        setConnectTimeout(connectTimeout, TimeUnit.MILLISECONDS)
    }

    fun setConnectTimeout(
        connectTimeout: Long?,
        unit: TimeUnit
    ) {
        this.connectTimeout = getTimeOut(connectTimeout, unit)
    }

    fun getReadTimeout(): Long? {
        return readTimeout
    }

    fun setReadTimeout(readTimeout: Long?) {
        setReadTimeout(readTimeout, TimeUnit.MILLISECONDS)
    }

    fun setReadTimeout(
        readTimeout: Long?,
        unit: TimeUnit
    ) {
        this.readTimeout = getTimeOut(readTimeout, unit)
    }

    fun getWriteTimeout(): Long? {
        return writeTimeout
    }

    fun setWriteTimeout(writeTimeout: Long?) {
        setWriteTimeout(writeTimeout, TimeUnit.MILLISECONDS)
    }

    fun setWriteTimeout(
        writeTimeout: Long?,
        unit: TimeUnit
    ) {
        this.writeTimeout = getTimeOut(writeTimeout, unit)
    }

    private fun getTimeOut(
        timeout: Long?,
        unit: TimeUnit
    ): Long {
        var resultTime: Long = 0
        when (unit) {
            TimeUnit.NANOSECONDS -> resultTime =
                unit.convert(timeout!!, TimeUnit.NANOSECONDS)
            TimeUnit.MICROSECONDS -> resultTime =
                unit.convert(timeout!!, TimeUnit.MICROSECONDS)
            TimeUnit.MILLISECONDS -> resultTime =
                unit.convert(timeout!!, TimeUnit.MILLISECONDS)
            TimeUnit.SECONDS -> resultTime =
                unit.convert(timeout!!, TimeUnit.SECONDS)
            TimeUnit.MINUTES -> resultTime =
                unit.convert(timeout!!, TimeUnit.MINUTES)
            TimeUnit.HOURS -> resultTime =
                unit.convert(timeout!!, TimeUnit.HOURS)
            TimeUnit.DAYS -> resultTime =
                unit.convert(timeout!!, TimeUnit.DAYS)
            else -> {
            }
        }
        return resultTime
    }
}